src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -389,16 +392,30 @@ private async Task<long> InsertGlobalStateTableRowAsync(SqlConnection connection
                 }
             }
 
-            string insertRowGlobalStateTableQuery = $@"
+            bool migrateOldUserFunctionIdData = await this.OldUserFunctionIdRecordExists(connection, transaction, userTableId, cancellationToken);
```

Comment:
Do we need to do this check? Couldn't you just make each of the below queries do the check regardless and migrate as needed? Would save a bit of execution time and duplication of the query.

Reply:
The reason I added the method is to add a telemetry event to track migration which I totally missed to add here. Will look into if there is a way to not have this check and still track migration.

Reply:
Ok, although given how little we look at this stuff currently I'd question whether we actually care about tracking this. Is that something you see being valuable?

Anyways it's fine if you want to keep this, not a huge deal (and we do other similar things elsewhere anyways).

Reply:
I don't really think that tracking is required here but definitely useful when deciding on deprecation path, we can look if there are any still using the migration process and take a call. Plus Chris asked me to add a telemetry point :)

Reply:
We could just look at version numbers for that, right? It's not like people can opt out of this so as long as everyone is on the version with this change or higher we should be able to assume they've migrated